packages/nimble-components/src/table/specs/README.md

```diff
@@ -133,13 +133,25 @@ The various APIs/features of the `nimble-table` will be split up amongst several
     -   Define how the action menu gets associated with a particular column
     -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
 
+_Attributes_
+
+-   `id-field-name` - An optional string attribute that specifies the field name within a row's record to use as a row's ID. If the attribute is not specified, a default ID will be generated. If the attribute is invalid, no rows in the table will be rendered, and the table will enter an invalid state according to the `validity` property and `checkValidity()` function. The attribute is invalid in the following conditions:
+    -   Multiple records were found with the same ID
+    -   A record was found that did not have a field with the name specified by `id-field-name`
+    -   A record was found where `id-field-name` did not refer to a value of type `string`
+
 _Properties_
 
-Placeholder
+-   `data` - An array of key/value pairs where each item in the array represents one row of data. For more information about the `data` property, refer to the [data API spec](table-data-api.md).
+-   `validity` - Readonly object that represents the validity states that the table's configuration can be in.
```

Comment:
Should we describe the high-level structure of this now or add more later? I'd at least initially advocate for a readonly object of boolean values and the object has type `TableValidityState` to mirror [ValidityState](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState)

Reply:
I wasn't sure what the right level of detail to put here was. I've updated it as you suggested.